### PR TITLE
Move `world_format.txt` to `world_format.md` and refactor

### PR DESCRIPTION
**Pull request by Bituvo**
This PR moves `world_format.txt` to a Markdown file (like what happened with `lua_api.txt`) and completely refactors it. I believe it is a lot more readable and easy to add to now. If I got anything wrong, please let me know in a review!

### CODE REVIEW

File: doc/world_format.md [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+    Empty
+    Empty
+    Empty
+    Empty
+    Empty
+    Empty
+    EndInventoryList
+    EndInventory
+
+# Table of Contents
```
Comment (appgurueu) [inline doc/world_format.md:620]: I feel like such a table of contents is (1) redundant and (2) probably will eventually be out-of-sync with the entire document. I think generating a ToC is the job of Markdown renderers.

Comment (rubenwardy) [reply doc/world_format.md:620]: GitHub shows a table of contents in the top left

Comment (Bituvo) [reply doc/world_format.md:620]: Ah. I didn't know that, I'll remove them

**Comment by Bituvo** [comment]
Resolved merge conflicts.

**Comment by Montandalar** [comment]
I would like to ask about the logic of leaving the .txt files behind. Personally I've taken to deleting the new lua_api.txt in my copy because it cripples the speed of tab completion.

It's also my personal opinion that whitespace changes and touching files just to fix the vanity of code style is a waste of everybody's time, and often used for commit number/resume padding. That's my opinion and you're entitled to your own.

**Comment by Bituvo** [comment]
> I would like to ask about the logic of leaving the .txt files behind. Personally I've taken to deleting the new lua_api.txt in my copy because it cripples the speed of tab completion.

That's what rubenwardy did when they switched the Lua API to Markdown (#12449), so I decided to stay consistent. I would assume that the pointer .txt files will be deleted at some point.
> It's also my personal opinion that whitespace changes and touching files just to fix the vanity of code style is a waste of everybody's time, and often used for commit number/resume padding. That's my opinion and you're entitled to your own.

When a PR is merged, all of the changes in the forked branch are squashed into one. Each commit I've made for this branch will be combined into one, with the PRs title. Additionally, lots of people read the documentation when they are developing something for the engine, and readability and organization are very important parts of that.

**Comment by rubenwardy** [comment]
You don't need to add a new .txt for world_format, it's only needed for lua_api.txt as it's a highly linked to file

**Comment by Bituvo** [comment]
I'll delete it.

**Review by sfan5 — APPROVED** [review]

File: doc/world_format.md [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+      `name` VARCHAR(32) UNIQUE,
+      `password` VARCHAR(512),
+      `last_login` INTEGER
+    );
+    CREATE TABLE `user_privileges` (
+      `id` INTEGER,
+      `privilege` VARCHAR(32),
+      PRIMARY KEY (id, privilege)
+      CONSTRAINT fk_id FOREIGN KEY (id) REFERENCES auth (id) ON DELETE CASCADE
+    );
```
Comment (srifqi) [inline doc/world_format.md:86]: Can we have SQL syntax highlighting here?

```sql
CREATE TABLE `auth` (
    `id` INTEGER PRIMARY KEY AUTOINCREMENT,
    `name` VARCHAR(32) UNIQUE,
    `password` VARCHAR(512),
    `last_login` INTEGER
);
CREATE TABLE `user_privileges` (
    `id` INTEGER,
    `privilege` VARCHAR(32),
    PRIMARY KEY (id, privilege)
    CONSTRAINT fk_id FOREIGN KEY (id) REFERENCES auth (id) ON DELETE CASCADE
);
```

File: doc/world_format.md [hunk 3]
```diff
@@ -0,0 +1,616 @@
...
+
+Today we know that `SQLite3` allows multiple primary keys (which would allow
+storing coordinates separately), but the format has been kept unchanged for
+that part.
+
+## `map.sqlite`
+`map.sqlite` is a `SQLite3` database, containing a single table, called
+`blocks`. It looks like this:
+
+    CREATE TABLE `blocks` (`pos` INT NOT NULL PRIMARY KEY,`data` BLOB);
```
Comment (srifqi) [inline doc/world_format.md:278]: ... and also here?

```sql
CREATE TABLE `blocks` (`pos` INT NOT NULL PRIMARY KEY,`data` BLOB);
```

**Review by srifqi — APPROVED** [review]
I found no major (content) changes from `.txt` to `.md`.

**Comment by srifqi** [comment]
Sorry for hitchhiking this PR. I added some small fixes to be included in these changes.

**Comment by Bituvo** [comment]
It would seem as though your changes broke the PR 😅

**Comment by srifqi** [comment]
That is weird. It failed when installing dependencies, though, so the failure is not from this PR specifically.

**Pull request merged**